### PR TITLE
GameSelectorModel.java - Couple cleanups

### PR DESCRIPTION
**Pull request by DanVanAtta**
- fix variable names, remove 'm_' prefix
- reduce method scope where possible
- remove always true boolean condition
- convert always false variable parameter to a 'hardcoded' false so it's
more clear the value is always false at that point of execution
- add a class javadoc

### CODE REVIEW

File: src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java [hunk 1]
```diff
@@ -25,40 +25,38 @@
...
+  private ClientModel clientModelForHostBots;
 
   public GameSelectorModel() {
     setGameData(null);
-    m_fileName = "";
   }
 
   public void resetGameDataToNull() {
     setGameData(null);
-    m_fileName = "";
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java]: Is this statement safe to remove?  It doesn't appear that `setGameData()` resets `fileName` to an empty string.

File: src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java [hunk 2]
```diff
@@ -254,17 +250,17 @@ public void loadDefaultGame(final boolean forceFactoryDefault) {
     // we don't want to load a game file by default that is not within the map folders we can load. (ie: if a previous
     // version of triplea
     // was using running a game within its root folder, we shouldn't open it)
-    NewGameChooserEntry selectedGame = null;
+    NewGameChooserEntry selectedGame;;
```
Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java]: Static analysis: "Unnecessary semicolon"

File: src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java (hunk 1 above)
Comment (DanVanAtta) [reply src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java]: Good catch, likely not, I think I thought this was a constructor.

**Comment by ssoloff** [comment]
@DanVanAtta You accidentally restored `m_fileName = ""` instead of `fileName = ""`.

**Comment by DanVanAtta** [comment]
dang.. this branch got messed up. Closing.

**Pull request closed without merging**